client/state/jetpack-connect/actions.js

```diff
@@ -196,30 +196,60 @@ export function retryAuth( url, attemptNumber ) {
 	};
 }
 
+export function createSocialAccount( socialInfo ) {
+	return dispatch => {
+		dispatch( recordTracksEvent( 'calypso_jpc_social_create_account' ) );
+
+		dispatch( { type: JETPACK_CONNECT_CREATE_ACCOUNT } );
+
+		/**
+		 * @TODO (sirreal) update to `jetpack-connect` when D11099-code lands
+		 *
+		 * The signup flow is required and affects some post signup activity.
+		 * `account` should be safe until patch lands.
+		 */
+		createSocialUser( socialInfo, 'account' /* 'jetpack-connect' */ ).then(
+			( { username, bearerToken } ) => {
+				dispatch( recordTracksEvent( 'calypso_jpc_social_createaccount_success' ) );
+				dispatch( {
+					type: JETPACK_CONNECT_CREATE_ACCOUNT_RECEIVE,
+					userData: { username },
+					data: { bearer_token: bearerToken },
```

Comment:
Provide same shape of data as `createAccount`, consumed:

https://github.com/Automattic/wp-calypso/blob/b3df26f1faecf6e847fc2c622631fd44dc260329/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L91-L92

`userData` uses `username` property:

https://github.com/Automattic/wp-calypso/blob/b3df26f1faecf6e847fc2c622631fd44dc260329/client/jetpack-connect/signup.js#L95-L99
https://github.com/Automattic/wp-calypso/blob/b3df26f1faecf6e847fc2c622631fd44dc260329/client/jetpack-connect/signup.js#L154